packages/node-resolve/src/index.js

```diff
@@ -295,11 +295,8 @@ export default function nodeResolve(options = {}) {
       // ignore IDs with null character, these belong to other plugins
       if (/\0/.test(importee)) return null;
 
-      const basedir = importer ? dirname(importer) : process.cwd();
-
-      if (shouldDedupe(importee)) {
-        importee = join(process.cwd(), 'node_modules', importee);
-      }
+      const rootDir = process.cwd();
+      const basedir = importer && !shouldDedupe(importee) ? dirname(importer) : rootDir;
```

Comment:
I think `!shouldDedupe` is a little hard to read, and `importer != null` is a bit easier to grok. Potentially rewrite the condition as:
```suggestion
      const basedir = !importer || shouldDedupe(importee) ? dirname(importer) : rootDir;
```

Reply:
Makes sense!